activestorage/app/models/active_storage/blob.rb

```diff
@@ -121,7 +121,11 @@ def generate_unique_secure_token(length: MINIMUM_TOKEN_LENGTH)
   # Returns a signed ID for this blob that's suitable for reference on the client-side without fear of tampering.
   # It uses the framework-wide verifier on <tt>ActiveStorage.verifier</tt>, but with a dedicated purpose.
   def signed_id
-    ActiveStorage.verifier.generate(id, purpose: :blob_id)
+    ActiveStorage.verifier.generate(
+      id,
+      purpose: :blob_id,
+      expires_in: ActiveStorage.urls_expire_in
+    )
```

Comment:
The signed blob id is a permanent reference, not an access control mechanism.

Making the signed identifier expire means all references to Blobs expire, breaking compatibility with anything that references blobs. That includes putting a maximum lifetime of 5 minutes on all fragment caches that reference blob URLs.

Reply:
Thanks, see my response below: https://github.com/rails/rails/pull/38843#issuecomment-615150252